### PR TITLE
Adds Radial Menus to Cyborg Module and Sprite selection

### PR DESCRIPTION
**Pull request by PacifistDalek**
## What Does This PR Do
Resurrects PR #16769 by @SabreML

Replaces the jank-as-hell "Does this look good to you?" module/sprite selector with a lovely new radial menu. Plus you can back out of selection after you select your module but before you select your sprite if you change your mind purely based on aesthetics. The name of the module and sprite still show up as mouseover pop-up text for those who are new.

Also renames "Miner" to "Mining" in the module selection, so that it grammatically matches the other modules. This required renaming Miner to Mining on the custom borg sprites, and i cleaned up one unused duplicate sprite state in custom borg sprites as per SabreML's original PR.

Created new module selection sprites (as that was the last thing left on the original PR) to represent each job.

## Why It's Good For The Game
borg drip (part 1)

## Images of changes
**Module Selection:**
Screenshot 2022-10-01 121432
**Sprite Selection:**
Screenshot 2022-10-01 121453
Screenshot 2022-10-01 121534
**Module Selection Sprites:**
image

## Testing
Tried choosing various modules and sprites, tested xeno borg, combat borg, and custom sprites as well.

## Changelog
:cl: SabreML, PacifistDalek
add: Added a radial menu system for cyborg module selection.
tweak: Renamed the "Miner" cyborg module to "Mining"
/:cl:

### CODE REVIEW

**Comment by Christasmurf** [comment]
Will sec module use the same as the combat module? While currently only available to be picked if VV'd by admins, it still requires you to actually choose the module rather than just making you a secborg straight away

**Comment by PacifistDalek** [comment]
> Will sec module use the same as the combat module? While currently only available to be picked if VV'd by admins, it still requires you to actually choose the module rather than just making you a secborg straight away
> 
> Edit: you can also choose it if enabled for red ERTs

that’s a very good question! it should but i haven’t tested that, i’ll make sure to do that

**Comment by PacifistDalek** [comment]
I’m not sure how to test the sec module because i don’t know how admins VV them in, if anyone does, please share!

**Comment by farie82** [comment]
> I’m not sure how to test the sec module because i don’t know how admins VV them in, if anyone does, please share!

I'm unsure how admins would do it. But I know how to do it in code.
image
Go there and add `"Security"` to the list. It is in the `code\modules\mob\living\silicon\robot\robot.dm` file.

I'm converting this PR to a draft for now. Feel free to convert it back over when you've finished/tested the security module

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -306,132 +308,202 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
...
+  * Returns a list of choosable module types, associated with the module icon for the radial menu.
+  *
+  * Key: Module name | Value: Module 'icon'
+  *
+  * By default this returns the Engineering, Janitor, Medical, Mining, and Service modules.
+  * If there are any [/mob/living/silicon/robot/var/force_modules] set, then they are returned instead.
+  * If the MMI has a xenomorph brain in it ([/obj/item/mmi/var/alien]), then only the "Hunter" module is returned.
+  */
+/mob/living/silicon/robot/proc/get_module_types()
+	var/list/standard_modules = list(
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/robot/robot.dm:334]:
```suggestion
	var/static/list/standard_modules = list(
```
unless these lists get changed, they can be static lists

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -306,132 +308,202 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
...
+		"Engineering" = image('icons/mob/robots.dmi', "engi-radial"),
+		"Janitor" = image('icons/mob/robots.dmi', "jan-radial"),
+		"Medical" = image('icons/mob/robots.dmi', "med-radial"),
+		"Mining" = image('icons/mob/robots.dmi', "mining-radial"),
+		"Service" = image('icons/mob/robots.dmi', "serv-radial"))
+	var/list/special_modules = list(
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/robot/robot.dm:340]:
```suggestion
	var/static/list/special_modules = list(
```

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 3]
```diff
@@ -306,132 +308,202 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
...
+
+/**
+  * Sets up the module items and sprites for the cyborg module chosen in `pick_module()`.
+  *
+  * Arguments:
+  * * selected_module - The name of the module chosen by the player in the previous procs.
+  * * selected_sprite - The name of the sprite chosen by the player in the previous procs.
+  * * module_sprites - The list of sprites possible for the given module. Used to transfer the `icon` and `icon_state` variables to the player.
+  */
+/mob/living/silicon/robot/proc/initialise_module(selected_module, selected_sprite, list/module_sprites)
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/robot/robot.dm:450]: this should probably use `initialize` instead of `initialise`

Comment (PacifistDalek) [reply code/modules/mob/living/silicon/robot/robot.dm]: i just changed the word, this won't require any more edits to make initialize work, will it? Just checking :)

**Review by S34NW — APPROVED** [review]
seems fine now, a lot going on here though

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by farie82 — APPROVED** [review]

**Pull request merged**